### ISSUE TITLE
Record and analyze console log

### ISSUE THREAD

**Issue by mkorpela** [MarketSquare/robotframework-browser#745]
**Is your feature request related to a problem? Please describe.**
Many frameworks such as React and used open source components will log warnings or errors to console when used wrong.
Also crash that will not destroy the whole page will show up in the console.
There is a lot of information here showing obvious bugs and problems.

**Describe the solution you'd like**
Console Logs should be somehow propagated to RF log. Also automatic fails for example from detected stacktrace might be good option.

**Comment by mkorpela** [comment]
Related to https://github.com/MarketSquare/robotframework-browser/issues/532

**Comment by mkorpela** [comment]
https://github.com/MarketSquare/robotframework-browser/issues/532

**Comment by VilleSalonen** [comment]
This is an improved version of code by @mkorpela from #532. Original implementation just waited for 5 seconds before returning all console messages. My version waits for Playwright's page object to emit close. I haven't tested this exhaustively yet but it seems promising.

extension.js

```
async function pipeLogToRF(page) {
    let messages = [];
    page.on("console", (message) => {
        console.log(message);
        messages.push(message.text());
    });
    await waitForPageClose(page);
    return messages;
}

async function waitForPageClose(page) {
    return new Promise((resolve) => {
        page.on("close", () => resolve());
    });
}

exports.__esModule = true;
exports.pipeLogToRF = pipeLogToRF;
```

```
*** Settings ***
Library     Browser  jsextension=${CURDIR}/extension.js
Force Tags  focus

*** Test Cases ***
Example Test
    Open Browser  https://www.reaktor.com
    ${p}=  Promise To  pipeLogToRF
    Sleep  10
    Click  a[data-title="Work"]
    Take Screenshot
    Go to  https://www.hs.fi
    wait until network is idle
    Close Page
    ${data}=  wait for  ${p}
```

**Comment by mkorpela** [comment]
The PR is also related to this https://github.com/MarketSquare/robotframework-browser/pull/747/files

**Comment by VilleSalonen** [comment]
Sorry, I totally forgot that pull request even though I saw it myself before. It's vastly better than my attempt.

**Comment by aaltat** [comment]
Fixed in ##747
